src/PointwiseFunctions/AnalyticSolutions/Tags.hpp

```diff
@@ -80,18 +73,231 @@ struct Error : db::PrefixTag, db::SimpleTag {
 /// \see ::Tags::AnalyticSolutions
 struct AnalyticSolutionsBase : db::BaseTag {};
 
-/// The analytic solutions for all `FieldTags`
+namespace detail {
+template <typename Tag>
+struct AnalyticImpl : db::PrefixTag, db::SimpleTag {
```

Comment:
You use this tag quite a lot outside this file, so `detail` and `Impl` seems inappropriate. How about this tag remains `::Tags::Analytic`, so you can drop the renames in other files? Then the `std::optional` tag above could become `::Tags::AnalyticOptional`.

Reply:
It's only used a lot right now because the observers still compute errors locally. Once that's no longer the case it's only used inside this file and in `src/Elliptic/BoundaryConditions/AnalyticSolution.hpp` (I don't count tests because things need to be tested). I find the naming you suggested terribly confusing, because then you get things from the `DataBox` using `Tags::AnalyticOptional` while I would expect to use `Tags::Analytic`.

Reply:
- Don't worry about `Elliptic/BoundaryConditions/AnalyticSolution.hpp`. I'll refactor it to store its own factory-created solution object.
- I'm worried about placing a Variables in a DataBox tag, which can only be used with private/detail/impl tags to access the constituents. I guess it's fine if it lives only in the ObservationBox, and accessing the constituents happens only ever through the subitems. I think it would be good to document that, if that is your intention.

Reply:
- Sounds good :)
- Ah yes, good point. I'll document that unlike traditional Variables, this (and error) use different subitems than might be expected, why that is, and how the type traits `is_analytic_solutions` plays into all of that. I'll have the type traits point to the simple tags for the documentation.